annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+    mock_revision.failed_validation_pages = [1, 2]
+    mock_revision.validated = [1, 2]
+    return [mock_revision]
+
+
+def test_get_task_revisions(
+    mock_session: Mock, mock_task_revisions: List[AnnotatedDoc]
+):
+    tenant = "test_tenant"
+    job_id = 1
+    task_id = 1
+    file_id = 1
+    task_pages = [1]
+
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.order_by.return_value = mock_query
+    mock_query.all.return_value = mock_task_revisions
+
+    result = get_task_revisions(
+        mock_session, tenant, job_id, task_id, file_id, task_pages
+    )
+
+    mock_session.query.assert_called_once_with(AnnotatedDoc)
+    mock_query.all.assert_called_once()
+
+    assert len(result) == 1
+    assert result[0].pages == {"1": ["data1"]}
+    assert result[0].failed_validation_pages == [1]
+    assert result[0].validated == [1]
+
+
+def test_get_task_info(mock_session: Mock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    task_id = 1
+    tenant = "test_tenant"
+
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.first.return_value = mock_task
+
+    result = get_task_info(mock_session, task_id, tenant)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.first.assert_called_once()
+    assert result == mock_task
+
+
+@pytest.fixture
+def annotated_file_pages() -> List[int]:
+    return [1, 2, 3]
+
+
+def test_unblock_validation_tasks(
+    mock_session: Mock,
+    mock_task: ManualAnnotationTask,
+    annotated_file_pages: List[int],
+):
+    mock_unblocked_tasks = MagicMock()
+    mock_session.query.return_value.filter.return_value = mock_unblocked_tasks
+    mock_unblocked_tasks.all.return_value = [mock_task]
+
+    result = unblock_validation_tasks(
+        mock_session, mock_task, annotated_file_pages
+    )
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_session.query.return_value.filter.assert_called_once()
+    mock_unblocked_tasks.update.assert_called_once_with(
+        {"status": TaskStatusEnumSchema.ready},
+        synchronize_session=False,
+    )
+    assert result == [mock_task]
+
+
+def test_get_task_stats_by_id(mock_session):
+    task_id = 1
+    mock_stats = MagicMock(spec=AnnotationStatistics)
+
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value.first.return_value = mock_stats
+
+    result = get_task_stats_by_id(mock_session, task_id)
+
+    mock_session.query.assert_called_once_with(AnnotationStatistics)
+    mock_query.filter.assert_called_once()
+    mock_query.filter.return_value.first.assert_called_once()
+
+    assert result == mock_stats
+
+
+def test_add_task_stats_record_existing_stats(mock_session):
+    task_id = 1
```

Comment:
Same as the previous functions, we don't use  this local variable anywhere else in the test. Pass it as kwarg.